src/core/loader/get-registry.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * @copyright BBC 2020
+ * @author BBC Children's D+E
+ * @license Apache-2.0
+ */
+import { itemsRegistry, initRegistry } from "../../components/shop/item-registry.js";
+
+export const loadRegistry = (screen, config) => {
+    const registryKeys = getRegistryKeys(config);
+    if (!registryKeys.length) return;
+
+    registryKeys.forEach(key => {
+        loadToCache(screen, key);
+    });
+
+    screen.load.start();
+
+    screen.load.on("complete", () => {
+        registryKeys.forEach(key => {
+            initRegistry(key, screen.cache.json.get(`registry-${key}`));
+        });
+        console.log("BEEBUG: itemsRegistry", itemsRegistry);
+    });
+};
+
+const getRegistryKeys = config => {
+    const registryKey = item => item[1].registryKey;
+
+    return Object.entries(config).filter(registryKey).map(registryKey);
+};
+
+const loadToCache = (screen, key) => {
```

Comment:
I think, if you make this **const loadToCache = screen => key => {**

then you can do 

**registryKeys.forEach(loadToCache(screen);**

on line 13 above. Might be worth us changing that in the other bit of loader at some point...

Reply:
I don't think I've seen (or, at least, understood) that double arrow pattern, but it's currying isn't it? Lovely little bit of functional goodness!

Reply:
672e4a9